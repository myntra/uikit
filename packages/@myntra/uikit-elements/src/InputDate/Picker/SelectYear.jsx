import React from 'react'
import PropTypes from 'prop-types'

function SelectYear({ year, onYearSelect, ...props }) {
  let startYearValue = year - 10
  let endYearValue = year + 10
  const yearArray = []
  if (startYearValue < 0) {
    startYearValue = 0
    endYearValue = 20
  }
  for (let i = startYearValue; i < endYearValue; i++) {
    yearArray.push(i)
  }

  return (
    <div {...props}>
      <select value={year} onChange={event => onYearSelect(Number(event.target.value))}>
        {yearArray.map(item => (
          <option value={item} key={item}>
            {item}
          </option>
        ))}
      </select>
    </div>
  )
}

SelectYear.propTypes = {
  year: PropTypes.number.isRequired,
  onYearSelect: PropTypes.func.isRequired
}

export default SelectYear
