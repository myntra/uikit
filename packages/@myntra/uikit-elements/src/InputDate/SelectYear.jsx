import React from 'react'
import PropTypes from 'prop-types'
import styles from './InputDate.css' //eslint-disable-line

const YearComponent = ({ currentYear, onYearSelect }) => {
  let startYearValue = currentYear - 10
  let endYearValue = currentYear + 10
  const yearArray = []
  if (startYearValue < 0) {
    startYearValue = 0
    endYearValue = 20
  }
  for (let i = startYearValue; i < endYearValue; i++) {
    yearArray.push(i)
  }

  return (
    <div className={styles.selectWrapper}>
      <select value={currentYear} onChange={event => onYearSelect(Number(event.target.value))}>
        {yearArray.map(item => (
          <option value={item} key={item}>
            {item}
          </option>
        ))}
      </select>
    </div>
  )
}

YearComponent.propTypes = {
  currentYear: PropTypes.number.isRequired,
  onYearSelect: PropTypes.func.isRequired
}

export default YearComponent
