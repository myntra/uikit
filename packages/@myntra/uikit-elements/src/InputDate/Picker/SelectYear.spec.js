import React from 'react'
import { shallow } from 'enzyme'
import SelectYear from './SelectYear'

describe('render', () => {
  it('should render', () => {
    const wrapper = shallow(<SelectYear year={2018} onYearSelect={jest.fn()} title="year" />)
    const options = wrapper.find('option')

    expect(options).toHaveLength(20)
    expect(options.at(0).text()).toBe('2008')
    expect(options.at(19).text()).toBe('2027')
    expect(wrapper).toMatchSnapshot()
  })

  it('should render 0 year', () => {
    const wrapper = shallow(<SelectYear year={2} onYearSelect={jest.fn()} title="year" />)
    const options = wrapper.find('option')

    expect(options).toHaveLength(20)
    expect(options.at(0).text()).toBe('0')
    expect(options.at(19).text()).toBe('19')
    expect(wrapper).toMatchSnapshot()
  })
})

describe('action', () => {
  it('select', () => {
    const fn = jest.fn()
    const wrapper = shallow(<SelectYear year={2018} onYearSelect={fn} />)

    wrapper.find('select').simulate('change', { target: { value: '2017' } })

    expect(fn).toHaveBeenCalledWith(2017)
  })
})
