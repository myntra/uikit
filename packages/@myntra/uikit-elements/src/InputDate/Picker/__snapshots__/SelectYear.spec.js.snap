// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render 0 year 1`] = `
<div
  title="year"
>
  <select
    onChange={[Function]}
    tabIndex={-1}
    value={2}
  >
    <option
      key="0"
      value={0}
    >
      0
    </option>
    <option
      key="1"
      value={1}
    >
      1
    </option>
    <option
      key="2"
      value={2}
    >
      2
    </option>
    <option
      key="3"
      value={3}
    >
      3
    </option>
    <option
      key="4"
      value={4}
    >
      4
    </option>
    <option
      key="5"
      value={5}
    >
      5
    </option>
    <option
      key="6"
      value={6}
    >
      6
    </option>
    <option
      key="7"
      value={7}
    >
      7
    </option>
    <option
      key="8"
      value={8}
    >
      8
    </option>
    <option
      key="9"
      value={9}
    >
      9
    </option>
    <option
      key="10"
      value={10}
    >
      10
    </option>
    <option
      key="11"
      value={11}
    >
      11
    </option>
    <option
      key="12"
      value={12}
    >
      12
    </option>
    <option
      key="13"
      value={13}
    >
      13
    </option>
    <option
      key="14"
      value={14}
    >
      14
    </option>
    <option
      key="15"
      value={15}
    >
      15
    </option>
    <option
      key="16"
      value={16}
    >
      16
    </option>
    <option
      key="17"
      value={17}
    >
      17
    </option>
    <option
      key="18"
      value={18}
    >
      18
    </option>
    <option
      key="19"
      value={19}
    >
      19
    </option>
  </select>
</div>
`;

exports[`render should render 1`] = `
<div
  title="year"
>
  <select
    onChange={[Function]}
    tabIndex={-1}
    value={2018}
  >
    <option
      key="2008"
      value={2008}
    >
      2008
    </option>
    <option
      key="2009"
      value={2009}
    >
      2009
    </option>
    <option
      key="2010"
      value={2010}
    >
      2010
    </option>
    <option
      key="2011"
      value={2011}
    >
      2011
    </option>
    <option
      key="2012"
      value={2012}
    >
      2012
    </option>
    <option
      key="2013"
      value={2013}
    >
      2013
    </option>
    <option
      key="2014"
      value={2014}
    >
      2014
    </option>
    <option
      key="2015"
      value={2015}
    >
      2015
    </option>
    <option
      key="2016"
      value={2016}
    >
      2016
    </option>
    <option
      key="2017"
      value={2017}
    >
      2017
    </option>
    <option
      key="2018"
      value={2018}
    >
      2018
    </option>
    <option
      key="2019"
      value={2019}
    >
      2019
    </option>
    <option
      key="2020"
      value={2020}
    >
      2020
    </option>
    <option
      key="2021"
      value={2021}
    >
      2021
    </option>
    <option
      key="2022"
      value={2022}
    >
      2022
    </option>
    <option
      key="2023"
      value={2023}
    >
      2023
    </option>
    <option
      key="2024"
      value={2024}
    >
      2024
    </option>
    <option
      key="2025"
      value={2025}
    >
      2025
    </option>
    <option
      key="2026"
      value={2026}
    >
      2026
    </option>
    <option
      key="2027"
      value={2027}
    >
      2027
    </option>
  </select>
</div>
`;
